Обработка заказов

rnikbond opened the pull request.

rnikbond committed:
Добавлен клиент (bin), который рассылает заказы в систему лояльности и на сервер

rnikbond committed:
Реализация Stringer() для стукруты User переделана на builder

rnikbond committed:
Добавлен сервис для обновления статусов начисления баллов у заказов

rnikbond committed:
Реализована задача для обновления статусов заказов в репозитории из системы лояльности

rnikbond committed:
Добавлено обновление в репозитории начисленных баллов из системы лояльности

rnikbond committed:
Добавлен обработчик запроса \balance.
Добавлены методы в репозиторий для работы с начислениями пользователя.

rnikbond committed:
Исправлен statictest

rnikbond committed:
Исправлен statictest

rnikbond committed:
Добавлен отладочный вывод

rnikbond committed:
Исправлен порт по умолчанию

rnikbond committed:
Добавлен заголовок в response

rnikbond committed:
Исправлено добавление заголовка

rnikbond committed:
Обработчики запросов вынесены в отдельные файлы.
Работы с баллами системы лояльности вынесена в отдельный сервис.
Добавлен логгер.

rnikbond committed:
К предыдущему коммиту

rnikbond committed:
Исправлены тесты

rnikbond committed:
Исправлено чтение переменных среды

rnikbond committed:
Исправлен тег для для чтения адреса запуска сервера из переменных среды

rnikbond committed:
Исправлен HTTP код при запросе списка заказов, когда у пользоваеля нет заказов

rnikbond committed:
Исправлена проверка заголовков запросов

rnikbond committed:
Исправлено тип поля в таблице, которое хранит время создания заказа

rnikbond committed:
Исправлен формат даты и времени регистрации заказа при ответе на запрос списка заказов пользователя

rnikbond committed:
Инфо лог перенаправил в stdout

rnikbond committed:
Исправлен тип поля "Номер заказа" в ответе на запрос списка заказов пользователя

rnikbond committed:
Исправлено чтение адреса системы лояльности

rnikbond committed:
Исправлена ошибка в адресе к системе лояльности

rnikbond committed:
Исправлено преобразование количества баллов лояльности

rnikbond committed:
Исправлено преобразование количества баллов при запросе баланса пользователя

rnikbond committed:
Добавлен отладочный вывод

rnikbond committed:
Исправлено преобразование баллов при чтении их базы

rnikbond committed:
Убран отладочный вывод

rnikbond committed:
Доавблено использование context

rnikbond committed:
Добавлены тесты на сервисы: работа с заказама и работа с баллами лояльности

rnikbond committed:
Добавлены mock файлы для сервисов

rnikbond committed:
Тесты обработчиков запросов \login \register переределаны с использованием моков сервисов

rnikbond committed:
Добавлены тесты обработчика GET /orders

rnikbond committed:
Добавлены тесты обработчика POST /orders

rnikbond committed:
Добавлено закрытие Body в тесте

rnikbond committed:
Добавлен тест обработчика запроса /balance

ASPecherkin commented on internal/handlers/orders_list.go:
```diff
@@ -0,0 +1,44 @@
+		return
+	}
+
+	if len(orders) < 1 {
```
тут проще == 0 , потому что отрицательного числа тут быть не может

rnikbond replied:
Привычка избегать строгие равенства) Исправил

ASPecherkin commented on internal/repository/repository.go:
```diff
@@ -2,19 +2,32 @@
+	pkgOrder "gophermarket/pkg/order"
 )
 
 type Authorization interface {
```
тут скорее дело привычки - обычно разносят интерфейсы к тем бизнес моделям, где оно будет использоваться 
но тут как тебе удобнее

ASPecherkin commented on internal/tasks/scanner.go:
```diff
@@ -0,0 +1,185 @@
+
+func (scan LoyaltyScanner) Scan(ctx context.Context) error {
+
+	ticker := time.NewTicker(1 * time.Second)
```
лучше вынести это время в конфигурацию 
чтобы проще было в дальнейшем это менять

rnikbond replied:
Исправил

ASPecherkin commented on internal/tasks/scanner.go:
```diff
@@ -0,0 +1,185 @@
+		return pkgOrder.AccrualOrder{}, err
+	}
+
+	if len(orderAccrual.Status) < 1 {
```
тоже лучше == 0

rnikbond replied:
+

rnikbond replied:
Поправил

ASPecherkin reviewed (changes requested):
оставил несколько замечаний

rnikbond committed:
Исправлено по замечаниям ментора:
Интервал обращения к сервису лояльности вынесен в конфиг.
Сравнение длины строки/слайса теперь сравнивается с нулем.
